core/include/cib.h

```diff
@@ -71,12 +73,12 @@ static inline unsigned int cib_avail(cib_t *__restrict cib)
  *                      Must not be NULL.
  * @return index of next item, -1 if the buffer is empty
  */
-static inline int cib_get(cib_t *__restrict cib)
+static inline int16_t cib_get(cib_t *__restrict cib)
 {
-    unsigned int avail = cib_avail(cib);
+    uint16_t avail = cib_avail(cib);
 
     if (avail > 0) {
-        return (int) (cib->read_count++ & cib->mask);
+        return (int16_t) (cib->read_count++ & cib->mask);
```

Comment:
I'm unsure, if `sizeof(int16_t) == sizeof(uint16_t)` leads to problems here (though the original implementation obviously worked too).